### PR TITLE
Add retry limitations based on rawApiError

### LINKED ISSUE #2311: [1.33] Swap: app hangs on the price loading state when no liquidity
1. Open Swap page on Barn/Dev
2. Connect to Goerli
3. Select ETH/ZRX token pair, an amount to trade
4. Check the form

**AR**: the app hangs in loading price state (up to 1 minute)

**ER**: No liquidity error should be displayed without delays

### PR DESCRIPTION
**Pull request by tukantje**
# Summary

Fixes #2311

Relies on changes made in https://github.com/cowprotocol/cow-sdk/pull/120

As retry function has the error injected (i.e. not the whole response but just the body in this case) we need the enriched version, where we can extract this information.

# To Test

1. Open swap
2. Change to Goerli
3. Pick ETH/ZRX pair
4. Enter an amount
5. Low liquidity error should appear, without waiting a minute long

### CODE REVIEW

**Review by alfetopito — APPROVED** [review]

**Review by shoom3301 — APPROVED** [review]

**Comment by elena-zh** [comment]
Hey @tukantje , great job!
Some my notes: 
1. I noticed that the app hangs in nose picking cow state, and in a price loading state when there is no liquidity, and a user is on the Limit orders page
hang in limit orders
2. In this PR there is no tokens/tokens with balances/token lists for Gnosis chain. Could you pleae check why? 
image
image
image

Thanks!

**Review by alfetopito — APPROVED** [review]

**Comment by tukantje** [comment]
I've updated the cow-sdk to apply the same logic for native price endpoint, which should fix this.

**Comment by elena-zh** [comment]
Hey @tukantje , now the message LGTM when I select a token pair with insufficient liquidity on the Limit orders page. 
However, now I can get the app crash (white page) when
1. A token pair with insufficient liquidity is selected on Gorli (e.g. WETH-ZRX)
2. Then I change the network to Mainnet using network selector
image

Could you please take a look why this happens?

Also, I'm curious why I see this error in the console when I switch networks: 
image

Thanks

**Comment by tukantje** [comment]
Updated target to release branch.

File: src/cow-react/modules/limitOrders/hooks/useGetInitialPrice.ts [hunk 1]
```diff
@@ -9,8 +9,14 @@ import { parsePrice } from '@cow/modules/limitOrders/utils/parsePrice'
...
 import { useWalletInfo } from '@cow/modules/wallet'
 import { getNativePrice } from '@cow/api/gnosisProtocol'
+import { ApiError } from '@cowprotocol/cow-sdk'
 
-type PriceResult = number | Error | undefined
+type ErrorWithApiError = { errorType: string; description: string; rawApiError: ApiError }
+type PriceResult = number | Error | undefined | ErrorWithApiError
+
+function hasApiError(value: any): value is ErrorWithApiError {
+  return value && typeof value === 'object' && value.errorType && value.description && value.rawApiError
```
Comment (shoom3301) [inline src/cow-react/modules/limitOrders/hooks/useGetInitialPrice.ts:18]: Looks good for now. The change won't be needed after this:
https://github.com/cowprotocol/cow-sdk/pull/122/files#diff-7633654c6f288b7d31db55cbe6c975509378f0b8830b6fc0ffb694e3efc38b76R4

**Review by shoom3301 — APPROVED** [review]

**Review by elena-zh — APPROVED** [review]
LGTM!

**Pull request merged**